### PR TITLE
build: make sure to lift all akka dependencies to source when test-against-akka-master is used

### LINKED ISSUE #2747: sslconfig version 0.4.0 in Akka master
Tests are failing when running with latest Akka master because sslconfig was updated to 0.4.0.

```
Uncaught error from thread [ClientServerSpec-akka.actor.default-dispatcher-11]: com/typesafe/sslconfig/ssl/NoopHostnameVerifier, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[ClientServerSpec]
java.lang.NoClassDefFoundError: com/typesafe/sslconfig/ssl/NoopHostnameVerifier
	at com.typesafe.sslconfig.akka.AkkaSSLConfig.useJvmHostnameVerification(AkkaSSLConfig.scala:78)
	at akka.stream.scaladsl.TLS$.$anonfun$apply$2(TLS.scala:110)
	at akka.stream.impl.io.TLSActor.liftedTree1$1(TLSActor.scala:159)
	at akka.stream.impl.io.TLSActor.<init>(TLSActor.scala:159)
```

https://jenkins.akka.io:8498/job/akka-http-nightly/647/AKKA_VERSION=default,Node=akka-http-nightly-node,SCALA_VERSION=2.12.9,jdk=AdoptOpenJDK%208/consoleFull

I can't find that Akka HTTP is defining an explicit dependency to ssl-config so why doesn't it run with the transitive dependency from akka-stream?

### PR DESCRIPTION
**Pull request by jrudolph**
Fixes #2747. Might obsolete #2754.

### CODE REVIEW

**Comment by jrudolph** [comment]
Going ahead with merging to check if it fixes the nightlies.

File: build.sbt [hunk 1]
```diff
@@ -149,7 +149,7 @@ lazy val httpCore = project("akka-http-core")
   .addAkkaModuleDependency("akka-stream", "provided")
   .addAkkaModuleDependency("akka-stream-testkit", "test")
   .addAkkaModuleDependency(
-    "akka-stream",
+    "akka-stream-testkit",
```
Comment (patriknw) [inline build.sbt]: ok, but should the above dependency to `akka-stream-testkit` still be there?

Comment (jrudolph) [reply build.sbt]: Hmm good point, maybe it needs to be guarded with an opposite condition than the one below.

**Comment by jrudolph** [comment]
Anyway on Jenkins, the problem is still the same. I wonder why it worked on my machine.

File: build.sbt (hunk 1 above)
Comment (jrudolph) [reply build.sbt]: Seems it is ok like that because the other line will be evicted once https://github.com/akka/akka-http/pull/2761 is merged.

**Pull request merged**